rfd/0151-database-permission-management.md

```diff
@@ -43,12 +43,51 @@ expected values for object names are specified as a list, while the matching
 uses glob-like semantics, with `*` matching any set of characters. Empty list of
 names matches nothing, while an empty scope matches everything.
 
+The labels applied by the import rules can reference the object properties, like
+`schema` or `database`, to populate the labels with dynamic, object-dependent
+values.
+
 #### Examples
 
-The `widget-prod` import rule applies to databases with `env: prod` label. It
-has the priority 10. If there is a database object that matches the clauses in
-`spec.mappings.match`, the import rule will apply the specified labels to the
-database object (`env: prod`, `product: WidgetMaster3000`).
+The `import_all_objects` below is a catch-all import rule which tags all objects
+in all databases with values taken from database object spec. This also a
+default import rule that will be created on startup if there are no other rules
+present.
+
+```yaml
+kind: db_object_import_rule
+metadata:
+  name: import_all_objects
+  namespace: default
+spec:
+  database_labels:
+    - name: '*'
+      values:
+        - '*'
+  mappings:
+    - add_labels:
+        database: '{{obj.database}}'
+        kind: '{{obj.object_kind}}'
+        name: '{{obj.name}}'
+        protocol: '{{obj.protocol}}'
+        schema: '{{obj.schema}}'
+        service_name: '{{obj.database_service_name}}'
```

Comment:
nit: for consistency, maybe use `xxx_name` like in match and scope?
```suggestion
    - add_labels:
        database_name: '{{obj.database}}'
        kind: '{{obj.object_kind}}'
        object_name: '{{obj.name}}'
        protocol: '{{obj.protocol}}'
        schema_name: '{{obj.schema}}'
        database_service_name: '{{obj.database_service_name}}'
```
And instead of an object name, maybe we can also interpolate `table_name`, `view_name`. I don't have a strong opinion on this though.

Reply:
> nit: for consistency, maybe use xxx_name like in match and scope?

Sure, that works. This is just a template anyway, so users can modify as needed.

> And instead of an object name, maybe we can interpolate the name as `table_name`, `view_name`. I don't have a strong opinion on this though.

This is bit tricky: to make it general, we would have to support the templates for the key, too:
```yaml
`{{obj.object_kind}}_name`: `{{obj.name}}`
```

I don't think we want to implement such a feature without good rationale, seems too easy to misuse. We could do something similar with individual matching labels per object kind, but I think it is better to keep the template more  general.